src/clojure/nrepl/misc.clj

```diff
@@ -89,3 +90,28 @@
   []
   (.startsWith (System/getProperty "java.runtime.version")
                "1.8"))
+
+(def var-meta-whitelist
+  "A list of var metadata attributes are safe to return to the clients.
+  We need to guard ourselves against EDN data that's not encodeable/decodable
+  with bencode. We also optimize the response payloads by not returning
+  redundant metadata."
+  [:ns :name :doc :file :arglists :forms :macro :special-form
+   :protocol :line :column :added :deprecated :resource])
+
+(defn resolve-file
+  [path]
+  (or (some-> path io/resource str) path))
+
+(defn normalize-meta
```

Comment:
I wonder if this function shouldn't get a better name, now that it's getting promoted to a (more) public api.

Reply:
Probably it deserves a docstring now as well.

Reply:
`sanitize-meta`?